stl/inc/ratio

```diff
@@ -19,59 +20,54 @@ _STL_DISABLE_CLANG_WARNINGS
 #undef new
 
 _STD_BEGIN
-template <intmax_t _Val>
-struct _Abs : integral_constant<intmax_t, (_Val < 0 ? -_Val : _Val)> {}; // computes absolute value of _Val
-
-template <intmax_t _Ax, intmax_t _Bx, bool _Sfinae = false,
-    bool _Good = (_Abs<_Ax>::value <= INTMAX_MAX / (_Bx == 0 ? 1 : _Abs<_Bx>::value))>
-struct _Safe_mult : integral_constant<intmax_t, _Ax * _Bx> {}; // computes _Ax * _Bx without overflow
+_NODISCARD constexpr intmax_t _Abs(intmax_t _Val) { // computes absolute value sof _Val
+    return (_Val < 0 ? -_Val : _Val);
+}
 
-template <intmax_t _Ax, intmax_t _Bx, bool _Sfinae>
-struct _Safe_mult<_Ax, _Bx, _Sfinae, false> { // _Ax * _Bx would overflow
-    static_assert(_Sfinae, "integer arithmetic overflow");
-};
+_NODISCARD constexpr intmax_t _Safe_mult(intmax_t _Ax, intmax_t _Bx) { // computes _Ax * _Bx without overflow

```

Comment:
Top level on params `const` and `noexcept` function

Reply:
Done